core/cap-0033.md

```diff
@@ -467,51 +461,44 @@ Else if op.type() == UPDATE_SPONSORSHIP_SIGNER
         If op.signer().accountID != op.sourceAccount
             Fail with UPDATE_SPONSORSHIP_NOT_SPONSOR
     Else if a.ext.v1().ext.v() == 2
-        If a.ext.v1().ext.v2().signerSponsorships[i] != op.sourceAccount
+        If a.ext.v1().ext.v2().signerSponsoringIDs[i] != op.sourceAccount
             Fail with UPDATE_SPONSORSHIP_NOT_SPONSOR
         wasSponsored = true
 
-        reserve = a.ext.v1().ext.v2().signerSponsorships[i].reserve
         Load AccountEntry source with source.accountID == op.sourceAccount
-        If AvailableLimit(source, NATIVE) < reserve
-            Fail with UPDATE_SPONSORSHIP_LINE_FULL
-        source.balance += reserve
+        source.ext.v1().ext.v2().numSponsoring--
 
     If an account sID is sponsoring future reserves for Account(le)
         Load AccountEntry acc with acc.accountID == sID
         If AvailableBalance(acc, NATIVE) < baseReserve
-            Fail with UPDATE_SPONSORSHIP_UNDERFUNDED
-        acc.balance -= baseReserve
+            Fail with UPDATE_SPONSORSHIP_LOW_RESERVE
+        acc.ext.v1().ext.v2().numSponsoring++
 
-        Set acc.ext.v(1)
-        Set acc.ext.v1().ext.v(2)
-        acc.ext.v1().ext.v2().signerSponsorships[i].sponsoringID = s.sponsoringID
-        acc.ext.v1().ext.v2().signerSponsorships[i].reserve = baseReserve
+        Set a.ext.v(1)
+        Set a.ext.v1().ext.v(2)
+        a.ext.v1().ext.v2().signerSponsoringIDs[i].sponsoringID = s.sponsoringID
     Else
-        Load AccountEntry acc with acc.accountID == Account(le)
-        If AvailableBalance(acc, NATIVE) < requiredReserve
-            Fail with UPDATE_SPONSORSHIP_UNDERFUNDED
+        If AvailableBalance(a, NATIVE) < requiredReserve
+            Fail with UPDATE_SPONSORSHIP_LOW_RESERVE
 
         If wasSponsored
-            acc.numSponsored -= mult
-        Set acc.ext.v(1)
-        Set acc.ext.v1().ext.v(0)
+            a.ext.v1().ext.v2().numSponsoring--
+        Clear a.ext.v1().ext.v2().signerSponsoringIDs[i].sponsoringID
```

Comment:
It's not clear to me fro the changes in this pseudocode but will sponsor's funds used to cover the reserve of a sponsored account remain in the sponsors account? i.e. No funds will move, they just get locked up?

Reply:
Exactly, funds aren't moving at all anymore.